pkg/proxy/backing_image.go

```diff
@@ -0,0 +1,213 @@
+package proxy
+
+import (
+	"fmt"
+	"time"
+
+	grpccodes "google.golang.org/grpc/codes"
+	grpcstatus "google.golang.org/grpc/status"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/net/context"
+	"golang.org/x/sync/errgroup"
+	"google.golang.org/protobuf/types/known/emptypb"
+
+	spdkapi "github.com/longhorn/longhorn-spdk-engine/pkg/api"
+	spdkclient "github.com/longhorn/longhorn-spdk-engine/pkg/client"
+	rpc "github.com/longhorn/types/pkg/generated/imrpc"
+)
+
+const (
+	maxMonitorRetryCount         = 10
+	monitorRetryPollInterval     = 1 * time.Second
+	spdkTgtReadinessProbeTimeout = 60 * time.Second
+)
+
+func (p *Proxy) SPDKBackingImageCreate(ctx context.Context, req *rpc.SPDKBackingImageCreateRequest) (*rpc.SPDKBackingImageResponse, error) {
+	log := logrus.WithFields(logrus.Fields{
+		"name":             req.Name,
+		"backingImageUUID": req.BackingImageUuid,
+		"diskUuid":         req.DiskUuid,
+		"size":             req.Size,
+		"Checksum":         req.Checksum,
+		"FromAddress":      req.FromAddress,
+		"SrcLvsUuid":       req.SrcLvsUuid,
+	})
+
+	log.Info("Backing Image Server: Creating SPDk Backing Image")
+
+	if req.Name == "" || req.BackingImageUuid == "" || req.DiskUuid == "" || req.Checksum == "" || req.Size == 0 {
+		return nil, grpcstatus.Error(grpccodes.InvalidArgument, "failed to create spdk backing image due to parameters missing")
+	}
+
+	ret, err := p.spdkLocalClient.BackingImageCreate(req.Name, req.BackingImageUuid, req.DiskUuid, req.Size, req.Checksum, req.FromAddress, req.SrcLvsUuid)
+	if err != nil {
+		return nil, grpcstatus.Error(grpccodes.Internal, err.Error())
+	}
+	return spdkBackingImageToBackingImageResponse(ret), nil
+}
+
+func (p *Proxy) SPDKBackingImageDelete(ctx context.Context, req *rpc.SPDKBackingImageDeleteRequest) (*emptypb.Empty, error) {
+	log := logrus.WithFields(logrus.Fields{
+		"name":     req.Name,
+		"diskUuid": req.DiskUuid,
+	})
+	log.Info("Backing Image Server: Deleting SPDk Backing Image")
+	if req.Name == "" || req.DiskUuid == "" {
+		return nil, grpcstatus.Error(grpccodes.InvalidArgument, "failed to delete spdk backing image due to parameters missing")
+	}
```

Comment:
Use `validateProxyMethodParameters`

e.g.

```
	input := map[string]string{
		"engineName":     engineName,
		"volumeName":     volumeName,
		"serviceAddress": serviceAddress,
	}
	if err := validateProxyMethodParameters(input); err != nil {
		return nil, errors.Wrap(err, "failed to list replicas for volume")
	}
```

Reply:
removed the check